test/library/randomized_benchmarking/test_clifford_utils.py

```diff
@@ -221,3 +222,16 @@ def test_clifford_inverse_table(self):
         for lhs, rhs in enumerate(_CLIFFORD_INVERSE_2Q):
             c = compose_2q(lhs, rhs)
             self.assertEqual(c, 0)
+
+    @data(1, 2, 3, 4)
+    def test_clifford_synthesis(self, num_qubits):
+        """Check if clifford synthesis does not change Clifford"""
+        basis_gates = tuple(["rz", "h", "cz"])
+        coupling_tuple = (
+            None if num_qubits == 1 else tuple((i, i + 1) for i in range(num_qubits - 1))
```

Comment:
Perhaps it's worth to test also a non-linear connectivity (for 4 qubits).
What happens if the qubits are not near each other? do you get an error?

Reply:
Added such test cases at [3efc58a](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1288/commits/3efc58a374de9f9fe4e6f2b659505ad752b36fdc) and confirmed that it works well with non-linear (star and cycle) connectivities.
I don't understand what you mean by "the qubits are not near each other". Does it mean "disconnected" (e.g. ((0, 1), (2, 3))) or "not full" (e.g. (0, 2), (2, 3), (3, 4))? In both cases, `_synthesize_clifford` fails with TranspilerError: `The circuit has an invalid layout as two qubits need to interact in disconnected components of the coupling map...` or `Fewer qubits in the circuit (4) than the coupling map (5)...`. I think these are expected results for users.